src/fwriteR.c

```diff
@@ -5,18 +5,23 @@
 #define DATETIMEAS_EPOCH     2
 #define DATETIMEAS_WRITECSV  3
 
+static bool utf8=false;
+static bool native=false;
+#define TO_UTF8(s) (utf8 && NEED2UTF8(s))
+#define TO_NATIVE(s) (native && (s)!=NA_STRING && !IS_ASCII(s))
+#define ENCODED_CHAR(s) (TO_UTF8(s) ? translateCharUTF8(s) : (TO_NATIVE(s) ? translateChar(s) : CHAR(s)))
```

Comment:
note that `TO_UTF8` and `TO_NATIVE` both run `TO_ASCII` and `S==NA_STRING`, so I think there's some redundancy here. 

how about

```
s == NA_STRING ? CHAR(s) : ( IS_ASCII(s) ? CHAR(s) : ( native ? translateChar(s) : IS_UTF8(s) ? translateCharUTF8(s) : CHAR(s) ) )
```

based on this truth table (the other 22/32 combinations are impossible because `utf8` :arrow_right: `!native` and `IS_ASCII` :arrow_right: `!IS_NATIVE` (and _vice versa_) 

![Truth table of all possible IS_ASCII, IS_UTF8, s==NA_STRING, utf8, native values](https://user-images.githubusercontent.com/7606389/97515631-d329f400-1967-11eb-8e92-915d57aaa9d0.png)

Reply:
Yes, you're right but after a 2nd thought, I still prefer the current version. The reasons are:

1. In terms of speed, the original version doesn't do anything unnecessary. For the default encoding value "", checking two boolean values should be much faster than checking NA_STRING and IS_ASCII. 

1. In terms of clarity, despite there's a small redundancy, the current version is easier to be understood.